call.go

```diff
@@ -213,10 +213,19 @@ func invoke(ctx context.Context, method string, args, reply interface{}, cc *Cli
 			// RPC has completed or failed.
 			put func()
 		)
+		codec := c.codec
+		if codec == nil {
+			codec = cc.dopts.codec
+		}
+		contentSubtype := c.contentSubtype
+		if contentSubtype == "" {
+			contentSubtype = codec.String()
+		}
 		// TODO(zhaoq): Need a formal spec of fail-fast.
 		callHdr := &transport.CallHdr{
-			Host:   cc.authority,
-			Method: method,
+			Host:           cc.authority,
+			Method:         method,
+			ContentSubtype: contentSubtype,
```

Comment:
Ya I don't know, I think we should move this to the full version. I am worried too about breaking other existing libraries, but they would have to incorrectly handle `content-type` to break (which they may be doing unfortunately). Adding the subtype gives more information, and we need it for example for yarpc.